### ISSUE TITLE
Developing tutorials for Quantum Machine Learning

### ISSUE THREAD

**Issue by ManjulaGandhi** [qiskit-advocate/qamp-fall-22#21]
### Description

Aim of our project is to develop tutorials or teaching materials for Quantum Machine Learning concepts. Idea is to encourage students to work on QML and compare them with classical machine learning techniques.

### Deliverables

Develop tutorials for QML algorithms
Output will be in the form of Videos, PPTs, Jupyter notebooks, a journal/conference paper

### Mentors details

- Mentor 1
  - Name: Dr. Manjula Gandhi S
  - GitHub ID: ManjulaGandhi
  - What they do: Associate Professor, Coimbatore Institute of Technology

### Number of mentees

2

### Type of mentees

- Mentee 1 & 2
  - Required:
    - Python Knowledge
    - Knowledge in Classical Machine Learning
  - Nice to have:
    - Good Writing Skills
    - Presentation Skills

**Comment by bopardikarsoham** [comment]
Hello, @ManjulaGandhi 
Soham here. I'm interested in this project. Perhaps we can discuss more about this project?

**Comment by Christophe-pere** [comment]
Hello, 
I'm interested in this project. I'm currently working on a paper where we compare ML and QML algorithms based on different encoding processes. Could we schedule a call to discuss this further?

**Comment by jayakumarksrit** [comment]
Eagerly waiting to apply 🙂

**Comment by SanyaNanda** [comment]
Hi, I am Sanya. I am interested in this project. I have good documentation skills and follow the divio structure. I have also engaged with qml qiskit textbook.

**Comment by derwind** [comment]
Hi, this project looks interesting for me. I have been developing ML prototype applications, especially using deep learning, in my company for the past few years and am very interested in the possibilities of QML. Unfortunately, I knew very little about quantum computing last year and was unable to attend QGSS2021, so I have currently worked on the QGSS2021 contents that has been incorporated into the Qiskit Textbook as self-study ones.
I have been using Python since 2015 and also have experienced the transition from Python 2 to Python 3. I am a relatively technical blogger and have some experience technical presentions internally and externally.

**Comment by derwind** [comment]
Hello @ManjulaGandhi I submitted my application to this issue as 1st choice. What is the next step? I am looking forward to discuss each other. Thank you.

**Comment by ManjulaGandhi** [comment]
@derwind we shall have a discussion via slack.

**Comment by kavitha-ishu** [comment]
@GemmaDawson Hi

**Comment by kavitha-ishu** [comment]
@GemmaDawson Hi.. I'm Kavitha S S

**Comment by GemmaDawson** [comment]
Please add your Checkpoint 1 presentation materials.

**Comment by jayakumarksrit** [comment]
Hi all,

Please find checkpoint 1 presentation attached with this email.

Thanks in Advance 1

**Comment by jayakumarksrit** [comment]
[](url)
[# 21 Developing Tutorials for Quantum Machine Learning.pptx](https://github.com/qiskit-advocate/qamp-fall-22/files/9752204/21.Developing.Tutorials.for.Quantum.Machine.Learning.pptx)

**Comment by SanyaNanda** [comment]
Hi

I have already added the presentation

Yours sincerely
Sanya Nanda

**Comment by jayakumarksrit** [comment]
[# 21 Developing Tutorials for Quantum Machine Learning.pdf](https://github.com/qiskit-advocate/qamp-fall-22/files/9752332/21.Developing.Tutorials.for.Quantum.Machine.Learning.pdf)

**Comment by jayakumarksrit** [comment]
### Check point 2:

In this project work, main objective is developing tutorials for Quantum Machine Learning (QML). To accomplish this, our team trying to create tutorials in the forms of interactive notebook, blog, video and paper. So, we try to compare Support Vector Machine (SVM) concept both in classical version as well as quantum version. For the first checkpoint, we laid roadmap to achieve possible educational materials and supportive self-study materials.

**Our progress so far:**
We discussed about various important aspects about learning difficulties related to quantum concepts. Especially people from classical background towards quantum version without proper introduction and substantial knowledge. 

**_Our key areas of focus have included:_**  
•	Analysis of classical Support Vector Machines.
•	Analysis about Quantum Support Vector Machines.
•	Comparison from both classical and Quantum Support Vector Machine (QSVM) version using wine dataset. 

**We identified some of the potential problem areas such as:**
•	Lack of resources to do self-study.
•	No guided tutorials available in the view of classical into quantum version.
•	Available contents are either too generic or very specific in nature.

**Intermediate work done so far:**

1) Created an interactive python notebook for classical version of SVM which includes following features
o	reduced number of features according to number of PCA components
o	Pick training size number of samples from each distro
o	Plotted confusion matrix with the help of heat-map for evaluating the behaviour and understanding the effectiveness of a binary or categorical classifier

2) Created an interactive python notebook for Quantum version of SVM which includes reduce number of features to number of qubits. Ability to utilize feature map in quantum version to fully harness exponential speedup. 

**_Future work for final project show case:_**
•	Based on the real time feedback from learners before and after tutorial videos. Also, to record their improvements in learning Quantum Support Vector Machines
•	Full-fledged interactive python note book to incorporate all educational objectives as mentioned in this project work
In terms of our more technical tasks, we have continued to work on the following areas:
	Including colourful images and interactive web-elements to enhance learning experience.
Visualization part:
A.	Visualize PCA dim. reduced Wine dataset as shown in figure 1.
B.	Plotting decision regions with three features as shown in figure 2.
C.	Visualizing Accuracy on both training set and test set as illustrated in figure 3.
D.	Illustrated ROC curve (receiver operating characteristic curve) and AUC (Area under the ROC Curve) for SVM in figure 4.
E.	Represented feature map decomposition using QSVM in figure 5.
 
Figure 1 PCA dim Reduced Wine Dataset 
![Figure 1 PCA dim Reduced Wine Dataset](https://user-images.githubusercontent.com/45353283/200013282-3793b431-ec2e-4c02-bb50-11bcfa1171dc.jpg)
 
Figure 2 SVM on Wine Dataset
![Figure 2 SVM on Wine Dataset](https://user-images.githubusercontent.com/45353283/200015716-60cd830e-2f0e-4e53-83df-6691984ecdf1.jpg)

Figure 3 Accuracy on Training and Test set
![Figure 3 Accuracy on Training and Test set](https://user-images.githubusercontent.com/45353283/200015874-c580c4da-81f5-47c2-9b16-0f4217e3a15e.jpg)
Figure 4 Roc-AUC Curve for SVM
![Figure 4 Roc-AUC Curve for SVM](https://user-images.githubusercontent.com/45353283/200015980-09698c6a-8562-41ca-b850-37df0b94c914.jpg)
Figure 5 Feature Map decomposition
![Figure 5 Feature Map decomposition](https://user-images.githubusercontent.com/45353283/200016094-b4ea9da9-553b-41ee-894a-666ab0e25596.jpg)


@GemmaDawson @kavitha-ishu @ManjulaGandhi

**Comment by jayakumarksrit** [comment]
### Visualization Part 
[Visualization part.pdf](https://github.com/qiskit-advocate/qamp-fall-22/files/9931869/Visualization.part.pdf)

**Comment by GemmaDawson** [comment]
@jayakumarksrit & @kavitha-ishu - please upload your Final Showcase presentation materials, and if needed, update the project Title and/or description.

**Comment by jayakumarksrit** [comment]
[# 21 Developing Tutorials for Quantum Machine Learning.pdf](https://github.com/qiskit-advocate/qamp-fall-22/files/10265294/21.Developing.Tutorials.for.Quantum.Machine.Learning.pdf)
[svm_wineQuantum.ipynb.txt](https://github.com/qiskit-advocate/qamp-fall-22/files/10265302/svm_wineQuantum.ipynb)
[svm_wineClassical.ipynb.txt](https://github.com/qiskit-advocate/qamp-fall-22/files/10265304/svm_wineClassical.ipynb)

**Comment by jayakumarksrit** [comment]
Hi Gemma,

Uploaded Final showcase presentation materials in github.
project #21 Developing tutorials for Quantum Machine Learning

Thanks in Advance !

**Comment by GemmaDawson** [comment]
Congratulations on completing all the requirements for QAMP Fall 2022!! 🌟🌟🌟
